File: UrashimaServer/UrashimaServer/Database/Models/Report.cs
﻿using System.ComponentModel.DataAnnotations;
using UrashimaServer.Database.Models;

namespace UrashimaServer.Models
{
    public class Report
    {
        [Key]
        public int Id { get; set; }
        public string ReportType { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public bool ReportStatus { get; set; } = false;
        public string TreatmentProcess { get; set; } = string.Empty;
        public DateTime SubmissionDate { get; set; } = DateTime.Now;
        public ICollection<ReportImage>? Images { get; set; }

        public int? AdsBoardId { get; set; }
        public AdsBoard? AdsBoard { get; set; }

        public int? AdsPointId { get; set; }
        public AdsPoint? AdsPoint { get; set; }

        public int? LocationId { get; set; }
        public Location? Location { get; set; }
    }
}

File: org/libs/business-layer/src/services/entities/report.ts
import { IAdLocation, IAdLocationDetail, IAdsBoard } from './ads';

interface IReport {
  lat: number;
  lon: number;
  reportType: string;
  name: string;
  email: string;
  phone: string;
  content: string;
  images: {
    image: string;
  }[];
  reportStatus: boolean; // false: Chưa xử lý, true: Đã xử lý
  treatmentProcess: string; // chuỗi rỗng: Chưa xử lý chưa có response, không thì ngược lại
  submissionDate: string; // Ngày tháng năm giờ phút giây
}

export interface IAdReport extends IReport {
  adsBoardID: number;
  adsPointID: number;
  reportData: IAdsBoard;
}
export interface ILocationReport extends IReport {
  latitude: number;
  longitude: number;
  reportData: IAdLocationDetail;
}

export interface IOfficerReport extends IReport {
  id: number;
  address: string;
}

export interface IOfficerReportDetail extends IOfficerReport {
  adsBoard?: IAdsBoard;

  adsPoint?: IAdLocation;

  location?: Pick<IAdLocation, 'id' | 'longitude' | 'latitude' | 'address'>;
}
